File: src/main/java/com/explosion204/wclookup/model/entity/Toilet.java
package com.explosion204.wclookup.model.entity;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;

@Data
@Entity
@Table(name = "toilet")
public class Toilet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String address;
    private String schedule;
    private double latitude;
    private double longitude;
    private double rating;
    private boolean confirmed;

    @OneToMany(mappedBy = "toilet")
    private List<Review> reviews;
}

File: android/app/src/main/java/com/example/wclookup/core/model/Toilet.kt
package com.example.wclookup.core.model

import com.example.wclookup.core.constant.NameConstant.ADDRESS
import com.example.wclookup.core.constant.NameConstant.CONFIRMED
import com.example.wclookup.core.constant.NameConstant.ID
import com.example.wclookup.core.constant.NameConstant.LATITUDE
import com.example.wclookup.core.constant.NameConstant.LONGITUDE
import com.example.wclookup.core.constant.NameConstant.RATING
import com.example.wclookup.core.constant.NameConstant.SCHEDULE
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Toilet(
    @SerializedName(ID) val id: Long,
    @SerializedName(ADDRESS) @Expose val address: String,
    @SerializedName(SCHEDULE) @Expose val schedule: String,
    @SerializedName(LATITUDE) @Expose val latitude: Double,
    @SerializedName(LONGITUDE) @Expose val longitude: Double,
    @SerializedName(RATING) @Expose val rating: Double,
    @SerializedName(CONFIRMED) @Expose val confirmed: Boolean
)